### PR TITLE
9- Dublin Core metadata standards

### LINKED ISSUE #9: Dublin Core metadata standards
The repository enables users to apply basic Dublin Core metadata to its records, as well as more granular elements (e.g. to support multilingualism, FAIR-compliance,discipline-based, and regional metadata schemas)

### PR DESCRIPTION
**Pull request by A-Souhei**
## Description
Adds Dublin Core elements in datasets; and in dcat exports;

image
image

image
image
image

closes https://github.com/fjelltopp/zarr-ckan/issues/9
Relates https://github.com/fjelltopp/fjelltopp-infrastructure/pull/271

_N.B: There are a lot of elements; In order to filter them, I will take a bit of them to document myself more._

## Checklist

Put an `x` in the boxes that apply to this pull request (you can also fill these out after opening the pull request).
You may not need to check all boxes.

- [ ] The Jira ticket for this issue has been updated to "Ready to Review" or equivalent.
- [x] I have developed these changes in discussion with the appropriate project manager.
- [ ] My code follows the general Fjelltopp documentation (see Confluence).
- [ ] I have made corresponding changes to the Fjelltopp documentation (see Confluence).
- [ ] I have rebased this branch with master.
- [ ] New dependency changes have been committed.
- [ ] I have added automated tests that prove my fix is effective or that my feature works.
- [ ] New and existing tests pass locally with my changes.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own code.
- [x] I have assigned at least one reviewer.
- [ ] I have assigned at least one label to this PR: "patch", "minor", "major".

### CODE REVIEW

**Comment by ChasNelson1990** [comment]
@A-Souhei is this ready for review?

**Comment by A-Souhei** [comment]
Updates: 

- added a validator for date and temporal_start
- added section dublin core metadata and extended attribute

File: ckanext/zarr/helpers.py [hunk 1]
```diff
@@ -122,15 +122,12 @@ def month_formatter(month):
...
-        match = re.match(pattern, cleaned_value)
-        if match:
-            inside_content = match.group(0)
-            elements = [element.strip() for element in inside_content.split(',')]
-            return elements
-        else:
-            return ''
-
+def multiple_select_formatter(value):
+    from markupsafe import Markup
```
Comment (ChasNelson1990) [inline ckanext/zarr/helpers.py]: General style guidelines for python is to put all imports at the start of the file. Can we move these out please?

File: ckanext/zarr/helpers.py [hunk 2]
```diff
@@ -122,15 +122,12 @@ def month_formatter(month):
...
+    import html
+    decoded_value = html.unescape(str(value))
+    cleaned = decoded_value.strip('[]{}\'')
+    values = [v.strip() for v in cleaned.split(',')]
+    if '{' and '}' in str(values):
```
Comment (ChasNelson1990) [inline ckanext/zarr/helpers.py]: I think your boolean logic here is incorrect... this actually says `True and ( '}' in str(values) )`. I think you want `'{' in str(values) and '}' in str(values)` or similar.

File: ckanext/zarr/helpers.py [hunk 3]
```diff
@@ -122,15 +122,12 @@ def month_formatter(month):
...
+        return []
```
Comment (ChasNelson1990) [inline ckanext/zarr/helpers.py]: If you make this return `""` then...

> ... you don't need to modify this line.

File: ckanext/zarr/templates/package/snippets/additional_info.html [hunk 4]
```diff
@@ -30,13 +30,16 @@ <h4 class="schema-group">{{ _(group) }}</h4>
         <tbody>
           {% for field in fields %}
             {% set ns.value = fn.get_pkg_value(pkg_dict, field.field_name) | trim %}
-            {% if ns.value != '' and ns.value != '[]' %}
+            {% if field.preset == 'multiple_select' %}
+              {% set ns.value = h.multiple_select_formatter(ns.value) %}
+            {% endif %}
+            {% if ns.value != '' and ns.value != '[]' and ns.value != [] %}
```
Comment (ChasNelson1990) [inline ckanext/zarr/templates/package/snippets/additional_info.html]: > If you make this return `""` then...

... you don't need to modify this line.

Comment (ChasNelson1990) [reply ckanext/zarr/templates/package/snippets/additional_info.html]: Right?

File: ckanext/zarr/helpers.py (hunk 3 above)
Comment (ChasNelson1990) [reply ckanext/zarr/helpers.py]: In fact... if you return `""` do you need any of the changes you made in `additional_info.html`?

**Review by ChasNelson1990 — CHANGES_REQUESTED** [review]
~edit: Looks like I have just reviewed new commits since I last reviewed. Will quickly go over your other changes too.~

LGTM!

File: ckanext/zarr/assets/css/zarr.css [hunk 5]
```diff
@@ -143,3 +143,99 @@
...
+}
+
+.package-read #dataset-resources {
+    margin-top: 36px;
+}
+
+.package-read .module-content {
+   margin-top: 36px;
+}
+
```
Comment (ChasNelson1990) [inline ckanext/zarr/assets/css/zarr.css]: Why multiple empty lines at the end of the file?

File: ckanext/zarr/profiles.py [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+
+
+# Additional namespaces we might need
+ADMS = Namespace("http://www.w3.org/ns/adms#")
+PROV = Namespace("http://www.w3.org/ns/prov#")
+DQV = Namespace("http://www.w3.org/ns/dqv#")
+
+log = logging.getLogger(__name__)
+
+DCT = Namespace("http://purl.org/dc/terms/")
```
Comment (ChasNelson1990) [inline ckanext/zarr/profiles.py]: Why is this not with the other namespaces?

File: ckanext/zarr/profiles.py [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+    # Preservation Statement
+    if dataset_dict.get('preservation_statement'):
+        g.add((dataset_ref, ADMS.status, Literal(dataset_dict['preservation_statement'])))
+
+    return g
+
+
+class DublinCoreDCATProfile(RDFProfile):
+    """
+    An RDF profile for Dublin Core metadata aligned with DCAT
```
Comment (ChasNelson1990) [inline ckanext/zarr/profiles.py:55]: Where did you get this from? We should cite any upstream sources to make it easier to justify in the future.

Comment (A-Souhei) [reply ckanext/zarr/profiles.py:55]: @ChasNelson1990 The materials are from here : https://github.com/ckan/ckanext-dcat/tree/master/ckanext/dcat/profiles.
There are a lot of namespaces and we will need more time to get all of them right , the ones here are not definitive and absolutely need more verification time. Shall I remove this file for now?

Comment (A-Souhei) [reply ckanext/zarr/profiles.py:55]: Made another ticket to discuss about it.

**Comment by A-Souhei** [comment]
NB: profiles.py for DCAT is WIP and requires improvements / updates

**Review by ChasNelson1990 — APPROVED** [review]
Let's get this merged. Any remaining work should be ticketed and done as future PRs. Great work on all of this @A-Souhei

**Pull request merged**